src/main/cms/cms_menu_blackbox.c

```diff
@@ -184,9 +195,15 @@ static const void *cmsx_Blackbox_onEnter(displayPort_t *pDisp)
 
     cmsx_Blackbox_GetDeviceStatus();
     cmsx_BlackboxDevice = blackboxConfig()->device;
-
-    blackboxConfig_p_ratio = blackboxConfig()->p_ratio;
+    cmsx_BlackboxRate = blackboxConfig()->sample_rate;
     systemConfig_debug_mode = systemConfig()->debug_mode;
+    
+    const uint16_t PIDFreq = (uint16_t)pidGetPidFrequency();
+    if (PIDFreq > 1000) {
+        tfp_sprintf(cmsx_PIDFreq, "%01d.%02d%s", PIDFreq / 1000, (PIDFreq % 1000) / 10, "KHZ");
```

Comment:
Should be `tfp_sprintf(cmsx_PIDFreq, "%1d.%02d KHZ", PIDFreq / 1000, (PIDFreq % 1000))`:
- don't use `0` in the format string unless you want 0 padding to the left;
- the `/ 10` for the Hz part seems to be wrong;
- no need to string replace `KHZ`.

Reply:
I added the /10 to show only 2 decimal digits (as in the configurator). Should i remove this to show 3 digits?

Reply:
Ah, now I see - the way you put it is a bit hard to understand, which will make it hard to maintain.
Probably better if you go:
```
const uint16_t pidFreq = (uint16_t)pidGetPidFrequency();
    if (pidFreq > 1000) {
        tfp_sprintf(cmsx_pidFreq, "%01d.%02d%s", (pidFreq / 10) / 100, (pidFreq / 10) % 100, "KHZ");
[...]
```

(Also note the camelCasing.)